### PR TITLE
hda-dma: refactor xrun handling

### PR DESCRIPTION
**Pull request by tlauda**
Changes behaviour of HDA Link overruns and underruns handling.
Let's no longer stop the stream, but just report an error.
It might happen that just after the release buffer is still
full/not yet empty after the previous run, but it shouldn't
affect the data, since stream has been paused anyway.
Also adds additional bit to better reflect hardware registers.

Signed-off-by: Tomasz Lauda <tomasz.lauda@linux.intel.com>

### CODE REVIEW

File: src/drivers/intel/cavs/hda-dma.c [hunk 1]
```diff
@@ -50,7 +50,8 @@
 #define DGCS_GEN	BIT(26)
 #define DGCS_FWCB	BIT(23)
 #define DGCS_BSC	BIT(11)
-#define DGCS_BOR	BIT(10) /* buffer overrun */
+#define DGCS_BOR	BIT(10) /* buffer overrun (input streams) */
+#define DGCS_BUR	BIT(10) /* buffer underrun (output streams) */
```
Comment (lgirdwood) [inline src/drivers/intel/cavs/hda-dma.c]: Should both be bit 10 ?

Comment (tlauda) [reply src/drivers/intel/cavs/hda-dma.c]: @lgirdwood Yes, but have different names depends on the stream's direction. It's just cosmetic change.

Comment (lgirdwood) [reply src/drivers/intel/cavs/hda-dma.c]: ok, can you comment this otherwise it may confuse.

Comment (tlauda) [reply src/drivers/intel/cavs/hda-dma.c]: @lgirdwood Added additional comment.

**Comment by jajanusz** [comment]
false positives in CI, sof logger abi mismatch, unrelated

**Pull request merged**